core/src/main/java/org/sonarsource/sonarlint/core/ConnectedSonarLintEngineImpl.java

```diff
@@ -198,24 +198,37 @@ public GlobalUpdateStatus update(ServerConfiguration serverConfig, @Nullable Pro
         }
         start();
       }
-      return getGlobalContainer().getUpdateStatus();
+      return getGlobalContainer().getGlobalStorageStatus();
     } finally {
       rwl.writeLock().unlock();
     }
   }
 
   @Override
+  public GlobalStorageUpdateCheckResult checkIfGlobalStorageNeedUpdate(ServerConfiguration serverConfig, ProgressMonitor monitor) {
+    checkNotNull(serverConfig);
+    return withReadLock(() -> {
+      checkUpdateStatus();
+      ConnectedContainer connectedContainer = new ConnectedContainer(globalConfig, serverConfig);
+      try {
+        connectedContainer.startComponents();
+        return connectedContainer.checkForUpdate(new ProgressWrapper(monitor));
+      } finally {
+        try {
+          connectedContainer.stopComponents(false);
+        } catch (Exception e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1166)